### ISSUE TITLE
commandline should provide helper to print command when using `&&` and `||`

### ISSUE THREAD

**Issue by kindermoumoute** [fish-shell/fish-shell#6583]
## Environment
```fish
$ fish --version
$ echo $version
fish, version 3.0.2
3.0.2
$ uname -a
Darwin kinderpc 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64
$ echo $TERM
xterm-256color
$ sh -c 'env HOME=$(mktemp -d) fish'
Welcome to fish, the friendly interactive shell
```

## Issue
`commandline` acts weirdly when used with `&&` and `||` operators. E.g. when I'm typing the command `echo "plop" && mycmd world<cursor>`:
```fish
commandline
> echo "plop" && mycmd world
commandline --tokenize
> echo plop mycmd world
```

## Expected

Either `--tokenize` should return `mycmd world`, or `commandline` should provide a flag to filter out all command outside of my `<cursor>` context.

**Comment by faho** [comment]
```fish
commandline --current-process --tokenize
>mycmd world # if the cursor is after the `&&`
>echo "plog" # if it is before
```

**Comment by kindermoumoute** [comment]
Had no idea what was the concept of `process` in fish, thank you! :-)

**Comment by kindermoumoute** [comment]
Sorry I thought that fixed it for a moment, but no I still have the issue 🤔

**Comment by kindermoumoute** [comment]
It works when I do `echo "plop"; mycmd world`, but it looks like `&&` keep both commands in the same process.

**Comment by kindermoumoute** [comment]
`commandline --tokenize --current-job ` do the job! 🎉 

Thank you for the help @faho

**Comment by faho** [comment]
Ah, okay, it appears this was fixed in 3.1. Current process would probably be what you want - job goes across pipelines, so if it was

```fish
echo foo | cat && bar
```

You'd get "echo foo | cat" or "bar", while with process you'd get "echo foo" or "cat" or "bar".

**Comment by faho** [comment]
Okay, #6214.
